clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -8030,6 +8155,22 @@ void Sema::AddSYCLIntelPrivateCopiesAttr(Decl *D, const AttributeCommonInfo &CI,
           << CI << /*non-negative*/ 1;
       return;
     }
+
+    // Check attribute applies to field as well as const variables, local
+    // variables, non-static data members, and device_global variables.
+    if (const auto *VD = dyn_cast<VarDecl>(D)){
+     if (!(isa<FieldDecl>(D) ||
+          (VD->getKind() != Decl::ImplicitParam &&
+           VD->getKind() != Decl::NonTypeTemplateParm &&
+           VD->getKind() != Decl::ParmVar &&
+	   (VD->hasLocalStorage() ||
+	    isTypeDecoratedWithDeclAttribute<SYCLDeviceGlobalAttr>(VD->getType())
+            || VD->getType().isConstQualified())))) {
```

Comment:
Or based on the correction you mention below, remove the ConstQualified check completely?

Reply:
>>IntelPrivateCopies was previously IntelNonConstVar, so shouldn't this be !VD->getType().isConstQualified() ?

@premanandrao, this patch relaxes the restriction that permits the private_copies attribute when applied on const variables. Previously we had restriction that the private_copies attribute would work with local non-const variables only. So i added VD->getType().isConstQualified() instead of !VD->getType().isConstQualified() as per request from @artemrad

Reply:
Before this change, we were accepting that attribute on non-const variables only, and now we want to extend it to const-variables also.   So having a check that it is ConstQualified does not make sense.    You can just remove it.

Reply:
> Before this change, we were accepting that attribute on non-const variables only, and now we want to extend it to const-variables also. So having a check that it is ConstQualified does not make sense. You can just remove it.

Agreed.